src/pybind/mgr/volumes/fs/fs_util.py

```diff
@@ -104,7 +104,12 @@ def listdir_by_ctime_order(fs, path):
     ens_with_ctime = []
     for en in entry_names:
         d_path = os.path.join(path, en)
-        stb = fs.lstat(d_path)
+        try:
+            stb = fs.lstat(d_path)
+        except cephfs.ObjectNotFound:
```

Comment:
> and raise VolumeException 

No. Instead of aborting due to a missing directory, we want to continue and return list of dir entries that are present.

Reply:
I got it. I meant instead of using ObjectNotFound catch cephfs.Error and use filesystem errno like ENOENT that's more readable.

Reply:
I was replying to your idea of raising VolumeException here instead of ignoring it.

About catching cephfs.Error vs ObjectNotFound, I pefer latter because it's more succinct. I replied about this in other review comment too.

Reply:
It's about code readability across the mgr/volumes. Having it uniform across the module helps. Please change it.

Reply:
I agree with @kotreshhr. Any improvements to exception handling (especially, if it does not change functionality) is best done as a separate change.

Reply:
> Any improvements to exception handling (especially, if it does not change functionality) is best done as a separate change.

@vshankar I think you missed the point. Kotresh's point is not about doing it as a separate change. It is about replacing `except ObjectNotFound` by

```
except cephfs.Error as e:
    if e.args[0] == ENOENT:
        continue
    else:
        raise
```

I prefer former because it's more readable, compact and succinct.

Reply:
I didn't miss the point - I completely understood it. My argument is, if you are going to handle `ObjectNotFound` in mgr/volumes, let's clean up that up in a separate change altogether. For now, stick with catching `cephfs.Error` as @kotreshhr mentioned.

Again, both approaches are correct - it's just code uniformity. And sorry, code readability, compactness is subjective, otherwise, there would be just one way to write code :)

Reply:
Marking this as resolved.